src/mlpack/methods/neat/genome_impl.hpp

```diff
@@ -0,0 +1,523 @@
+        newTarget));
+    directedGraph[sourceID].emplace(newTarget, ConnectionGene(innovID,
+        1, sourceID, newTarget));
+  }
+  else
+  {
+    if (!directedGraph[sourceID][newTarget].Enabled())
+    {
+      directedGraph[sourceID][newTarget].Enabled() = true;
+      for (size_t j  = 0; j < connectionGeneList.size(); j++)
+      {
+        if (connectionGeneList[j].Source() == sourceID &&
+            connectionGeneList[j].Target() == newTarget)
+        {
+          connectionGeneList[j].Enabled() = false;
+          break;
+        }
+      }
+    }
+  }
+}
+
+// Add node.
+template <class ActivationFunction>
+void Genome<ActivationFunction>::AddNodeMutation()
+{
+  size_t i = 0;
+  while (connectionGeneList[i].Source() == 0)
+  {
+    i = arma::randi<arma::uvec>(1, arma::distr_param(0, (int)
+        (connectionGeneList.size() - 1)))[0];
+  }
+  size_t sourceID = connectionGeneList[i].Source();
+  size_t targetID = connectionGeneList[i].Target();
+  size_t newNodeID = nextNodeID++;
+  size_t innovID1, innovID2;
+
+  // Check if these mutations have been made. Else, add them to the buffer.
+  std::pair<size_t, size_t> key1 = std::make_pair(sourceID, newNodeID);
+  std::pair<size_t, size_t> key2 = std::make_pair(newNodeID, targetID);
+
+  if (mutationBuffer.find(key1) == mutationBuffer.end())
+  {
+    innovID1 = nextInnovID++;
+    mutationBuffer[key1] = innovID1;
+  }
+  else
+    innovID1 = mutationBuffer[key1];
+
+  if (mutationBuffer.find(key2) == mutationBuffer.end())
+  {
+    innovID2 = nextInnovID++;
+    mutationBuffer[key2] = innovID2;
+  }
+  else
+    innovID2 = mutationBuffer[key2];
+
+  // Add the first connection to the containers.
+  directedGraph[sourceID].emplace(newNodeID, ConnectionGene(innovID1, 1,
+      sourceID, newNodeID));
+  connectionGeneList.emplace_back(ConnectionGene(innovID1, 1,
+      sourceID, newNodeID));
+
+  // Add the second connection to the containers.
+  connectionGeneList.emplace_back(ConnectionGene(innovID2, 1,
+      newNodeID, targetID));
+  directedGraph.emplace(std::piecewise_construct,
+                        std::make_tuple(newNodeID),
+                        std::make_tuple());
+  directedGraph[newNodeID].emplace(targetID, ConnectionGene(innovID2, 1,
+      newNodeID, targetID));
+
+  // Remove the lost connection.
+  directedGraph[sourceID][targetID].Enabled() = false;
+  connectionGeneList[i].Enabled() = false;
+
+  // If the genome is acyclic, change the depths.
+  if (isAcyclic)
+  {
+    nodeDepths.push_back(nodeDepths[sourceID] + 1);
+
+    // If this is the case, the connection we are splitting is part of the
+    // longest path.
+    if (nodeDepths[targetID] == nodeDepths[sourceID] + 1)
+    {
+      nodeDepths[targetID]++;
+      Traverse(targetID);
+    }
+  }
+}
+
+// Delete connection.
+template <class ActivationFunction>
+void Genome<ActivationFunction>::DelConnMutation()
+{
+  size_t i = 0;
+  i = arma::randi<arma::uvec>(1, arma::distr_param(0,
```

Comment:
You could use:

https://github.com/mlpack/mlpack/blob/12a50a055ba7f69340598329bd146ee37bec110f/src/mlpack/core/math/random.hpp#L106